### ISSUE TITLE
Using enumerize with paper_trail

### ISSUE THREAD

**Issue by ramaboo** [brainspec/enumerize#263]
I was hoping someone from the enumerize team could help with this issue https://github.com/airblade/paper_trail/issues/906. Paper trail is a very popular gem and they just release version 6.0.x which breaks when using enumerize and we are trying to figure out a fix. Any help you could offer would be awesome as the paper_trail team does not really know the internals of enumerize.

Thanks
David

**Comment by lest** [comment]
@ramaboo can you please try the `type-serialization` branch?

``` ruby
gem 'enumerize', github: 'brainspec/enumerize', branch: 'type-serialization'
```

**Comment by ramaboo** [comment]
Yes. I had to manually clone the gem and build it from that branch to get it to work with paper_trails bug template but after that it worked perfectly.

What's new in the type-serialization branch? and more importantly is that going to be incorporated into the next version of enumerize?

We use both enumerize and paper_trail in a large production app so were trying to figure out what our best course of action going forward is because we don't want to be stuck on the 5.2.x version of paper_trail forever.

Also thanks for your help with this! We love enumerize and use it on almost every model we have.

**Comment by lest** [comment]
@ramaboo You can see the changes in https://github.com/brainspec/enumerize/pull/264. The changes are in the master branch now.

`paper_trail` uses `@klass.type_for_attribute(attr).serialize(val)` to store the attribute values. `enumerize` didn't support that before and now it has support.

I plan to make a release after additional testing with a couple of production Rails apps to ensure that the changes didn't cause any regressions.

By the way, you should be able to use the gem from GitHub without cloning and building it manually. You can use the following line in your `Gemfile`:

```ruby
gem 'enumerize', github: 'brainspec/enumerize'
```

**Comment by ramaboo** [comment]
Awesome thanks! Yeah normally that would work in a gemfile for some reason it didn't with the template paper_trail uses for it's bug reports. I suspect it's because it only loads a minimal environment or some other issue with rvm being weird.

Were doing a big release tonight but over the next couple days will switch our app to using master branch on the repo and paper_trail 6 and see how it goes :)

**Comment by jaredbeck** [comment]
> paper_trail uses @klass.type_for_attribute(attr).serialize(val) to store the attribute values. enumerize didn't support that before and now it has support.

Thanks Sergey! I think this `type_for_attribute` stuff is Sean's work on the rails' attribute API that he started in rails 4.2 and finished in 5.0.

**Comment by ramaboo** [comment]
Any chance of getting a new release of enumerize with this in it?

**Comment by lest** [comment]
@ramaboo done! I released 2.1.0.
